pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -26,10 +37,201 @@ def synchronize(remote_pk, repository_pk):
     """
     remote = RpmRemote.objects.get(pk=remote_pk)
     repository = Repository.objects.get(pk=repository_pk)
-    # base_version = RepositoryVersion.latest(repository)
 
     if not remote.url:
         raise ValueError(_('A remote must have a url specified to synchronize.'))
 
     log.info(_('Synchronizing: repository={r} remote={p}').format(
         r=repository.name, p=remote.name))
+
+    first_stage = RpmFirstStage(remote)
+    DeclarativeVersion(first_stage, repository).create()
+
+
+class RpmFirstStage(Stage):
+    """
+    First stage of the Asyncio Stage Pipeline.
+
+    Create a :class:`~pulpcore.plugin.stages.DeclarativeContent` object for each content unit
+    that should exist in the new :class:`~pulpcore.plugin.models.RepositoryVersion`.
+    """
+
+    def __init__(self, remote):
+        """
+        The first stage of a pulp_rpm sync pipeline.
+
+        Args:
+            remote (RpmRemote): The remote data to be used when syncing
+
+        """
+        self.remote = remote
+
+    async def __call__(self, in_q, out_q):
+        """
+        Build `DeclarativeContent` from the repodata.
+
+        Args:
+            in_q (asyncio.Queue): Unused because the first stage doesn't read from an input queue.
+            out_q (asyncio.Queue): The out_q to send `DeclarativeContent` objects to
+
+        """
+        async def parse_repodata(primary_xml_path, filelists_xml_path, other_xml_path):
```

Comment:
All of these inline defined functions don't have to be nested here. They should be moved to be a peer of `__call__()` not defined inside it. As long as the `async def` syntax is used to define the method (like on L#69) the method will return a coroutine which can be awaited.